microprofile-fault-tolerance/README.adoc

```diff
@@ -0,0 +1,433 @@
+////
+This guide is _ported_ from the main Quarkus repository
+https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
+////
+include::../shared-doc/attributes.adoc[]
+
+= microprofile-fault-tolerance: MicroProfile Fault Tolerance
+:author: Tomas Hofman, Radoslav Husar
+:level: Intermediate
+:technologies: MicroProfile, Fault Tolerance
+
+[abstract]
+The `microprofile-fault-tolerance` quickstart demonstrates how to use Eclipse MicroProfile Fault Tolerance in {productName}.
+
+One of the challenges brought by the distributed nature of microservices is that communication with external systems is
+inherently unreliable. This increases demand on resiliency of applications. To simplify making more resilient
+applications, {productName} contains an implementation of the MicroProfile Fault Tolerance specification.
+
+In this guide, we demonstrate usage of MicroProfile Fault Tolerance annotations such as `@Timeout`, `@Fallback`,
+`@Retry` and `@CircuitBreaker`. The specification also introduces `@Bulkhead` and `@Asynchronous` interceptor bindings not
+covered in this guide.
+
+== Prerequisites
+
+To complete this guide, you need:
+
+* less than 15 minutes
+* an IDE
+* JDK 1.8+ installed with `JAVA_HOME` configured appropriately
+* Apache Maven 3.5.3+
+
+== The Scenario
+
+The application built in this guide simulates a simple backend for a gourmet coffee e-shop. It implements a REST
+endpoint providing information about coffee samples we have on store.
+
+Let's imagine, although it's not implemented as such, that some of the methods in our endpoint require communication
+to external services like a database or an external microservice, which introduces a factor of unreliability.
+
+== Solution
+
+We recommend that you follow the instructions in the next sections and create the application step by step.
+However, you can go right to the completed example.
+
+== Creating the Maven Project
+
+First, we need a new project. Create a new project with the following command:
+
+// TODO clarify what archetype we can use here
+[source, subs=attributes+]
+----
+mvn archetype:generate \
+    -DarchetypeArtifactId=wildfly-jakartaee-webapp-archetype \
+    -DarchetypeGroupId=org.wildfly.archetype \
+    -DarchetypeVersion=18.0.0.Final
```

Comment:
I don't think you want to use 18.0.0.Final, as this version does not contain fault-tolerance, right?

Reply:
And perhaps parameterize these?

Reply:
Sure, but 19.0.0.Final will be available when this PR is merged.

Reply:
Are we sure? There aren't any for WildFly 9, 10, 11, 12, 13, 14, 15, 16, nor 17.

Reply:
Oh for f@#$ sake...

Reply:
Just an update from the call: we are going to use a generic archetype and add a section how to import a bom.

Reply:
Done.